src/Features/Core/Portable/IntroduceVariable/AbstractIntroduceParameterService.cs

```diff
@@ -145,18 +144,41 @@ private static async Task<(bool shouldDisplay, bool containsClassExpression)> Sh
                     }
                 }
 
-                // If expression contains this or base keywords then we do not want to refactor call sites
-                // that are not overloads/trampolines because we do not know if the class specific information
-                // is available in other documents.
-                if (variable.RawKind == syntaxKinds.ThisExpression || variable.RawKind == syntaxKinds.BaseExpression)
+                var operation = semanticModel.GetOperation(variable, cancellationToken);
```

Comment:
instead of calling GetOperation on each variable in this loop, do GetOperation(expression) and walk that.  Simpler and easier and prevents accidentally missing something.

Reply:
👍 changed